src/helpers/optionalStacking/test/common.spec.ts

```diff
@@ -1,17 +1,152 @@
+                localIdentifier: STACK,
+                items: props.stackBy ? [props.stackBy] : [],
+            },
+        ];
+    }
+
+    it("should keep config as is if buckets have more than 1 measure", () => {
+        const buckets: VisualizationObject.IBucket[] = getBuckets({
+            measures: [M1, M2],
+            viewBy: [A1],
+        });
+        const config: IChartConfig = {
+            stackMeasures: true,
+            stackMeasuresToPercent: true,
+        };
+        const sanitizedBucketsAndStackingConfig: ISanitizedBucketsAndStackingConfig = getSanitizedBucketsAndStackingConfig(
+            buckets,
+            config,
+        );
+        expect(sanitizedBucketsAndStackingConfig.config).toEqual(config);
+    });
+
+    it("should sanitized stacking config if buckets have one measure and no stackBy", () => {
+        const buckets: VisualizationObject.IBucket[] = getBuckets({
+            measures: [M1],
+            viewBy: [A1],
+        });
+        const config: IChartConfig = {
+            stackMeasures: true,
+            stackMeasuresToPercent: true,
+        };
+        const sanitizedBucketsAndStackingConfig: ISanitizedBucketsAndStackingConfig = getSanitizedBucketsAndStackingConfig(
+            buckets,
+            config,
+        );
+        expect(sanitizedBucketsAndStackingConfig.config).toEqual({
+            stackMeasures: true,
+            stackMeasuresToPercent: false,
+        });
+    });
+
+    it("should sanitized stacking config if buckets have one measure and one stackBy", () => {
+        const buckets: VisualizationObject.IBucket[] = getBuckets({
+            measures: [M1],
+            viewBy: [A1],
+            stackBy: A2,
+        });
+        const config: IChartConfig = {
+            stackMeasures: true,
+            stackMeasuresToPercent: true,
+        };
+        const sanitizedBucketsAndStackingConfig: ISanitizedBucketsAndStackingConfig = getSanitizedBucketsAndStackingConfig(
+            buckets,
+            config,
+        );
+        expect(sanitizedBucketsAndStackingConfig.config).toEqual(config);
+    });
+
+    it("should sanitized stacking config if buckets have 1 measure and isComputeRatio", () => {
+        const buckets: VisualizationObject.IBucket[] = getBuckets({
+            measures: [M1WithRatio],
+            viewBy: [A1],
+        });
+        const config: IChartConfig = {
+            stackMeasures: true,
+            stackMeasuresToPercent: true,
+        };
+        const sanitizedBucketsAndStackingConfig: ISanitizedBucketsAndStackingConfig = getSanitizedBucketsAndStackingConfig(
+            buckets,
+            config,
+        );
+        expect(sanitizedBucketsAndStackingConfig.config).toEqual({
+            stackMeasures: false,
+            stackMeasuresToPercent: false,
+        });
+    });
+
+    it("should sanitized computeRatio if buckets have more than 1 measures", () => {
+        // after this test, measure computeRatio will be reset
```

Comment:
what about creating const A1..An, M1..Mn in before each to have it fresh for each test case? It is safer in case someone will add new test case

Reply:
updated